File: Intersection of Two Arrays/Intersection_of_Two_Arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        Set<Integer> st = new HashSet<>();
        Set<Integer> st2 = new HashSet<>();
        for (int i : nums1) {
            st.add(i);
        }
        for (int i : nums2) {
            st2.add(i);
        }
        int[] res = new int[st2.size()];
        int idx = 0;
        for (int i : st2) {
            if (st.contains(i)) {
                res[idx++] = i;
            }
        }
        return Arrays.copyOf(res, idx);
    }
}

File: Intersection of Two Arrays/Intersection_of_Two_Arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        unordered_set<int> st,st2;
        for(auto i: nums1) {
            st.insert(i);
        }
        for(auto i: nums2) {
            st2.insert(i);
        }
        vector<int> ans;
        for(auto i: st2) {
            if(st.count(i)) ans.push_back(i);
        }
        return ans;
    }
};